src/palette/internal/paletteprovider.cpp

```diff
@@ -617,6 +616,22 @@ void PaletteProvider::init()
     });
 }
 
+void PaletteProvider::setFilter(const QString& filter)
+{
+    // Remove the model when there is no search text to return no results
+    // and thus speed up the opening of the palette search text box.
+    // Restore the model as soon as the search text is non-empty.
+    if (filter.length() > 0) {
```

Comment:
Hey bud, figured I'd mention some nit-pick that the in-team "might" care about, but just in case: since filter is a QString, you could say `!filter.isEmpty()` here, .... and a little below you can:

Reply:
Thanks, changed.